src/Microsoft.IdentityModel.Logging.Abstractions/IIdentityLogger.cs

```diff
@@ -0,0 +1,49 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System.Diagnostics.Tracing;
```

Comment:
>System.Diagnostics.Tracing [](http://example.com/codeflow?start=6&length=26)

Is this dependency worth it just for the enum? Doesn't MISE implement it's own clone and then it's very easy for a consumer to map (we can include an example in the main lib) Would be a shame if this blocked someone's adoption.

Reply:
If we were to remove the dependency on System.Diagnostics.Tracing.Event and introduce a custom enum (say LogSeverityLevel enum) then log level needs to be mapped from EventLevel -> LogSeverityLevel. Not sure if there could be a performance impact. Benchmarking may be worth it here.

Reply:
If the new enum uses the same numeric values, then the mapping would be as simple as a double cast:
(LogSererityLevel)(int)eventLevel

Reply:
(System.Diagnostics.Tracing)[https://www.nuget.org/packages/System.Diagnostics.Tracing/] has minimal dependencies similar to System.Collections[https://www.nuget.org/packages/System.Collections/] that TelemetryClient depends on. This will be kept as is unless there is a good argument against it.